src/merkle_set.rs

```diff
@@ -114,6 +116,22 @@ fn radix_sort(range: &mut [[u8; 32]], depth: u8) -> ([u8; 32], NodeType) {
                 (child_hash, child_type)
             }
         }
+    } else if depth == 255 {
+        // this is an edge case where we make it all the way down to the
+        // bottom of the tree, and split the last pair. This has the same
+        // effect as the else-block, but since we use u8 for depth, it would
+        // overflow
+        debug_assert!(range.len() > 1);
+        debug_assert!(left < range.len() as i32);
+        (
+            hash(
+                NodeType::Term,
+                NodeType::Term,
+                &range[0],
+                &range[left as usize],
+            ),
+            NodeType::MidDbl,
+        )
     } else {
         let (left_hash, left_type) = radix_sort(&mut range[..left as usize], depth + 1);
```

Comment:
Since we now have two implementations of `depth == 255`, it might be nice to add an assert to the `else` case to distinguish it, as well.

Reply:
Yes, I guess that's the only difference. Not needed :)